CRM/Contribute/BAO/Contribution.php

```diff
@@ -5454,4 +5454,45 @@ public static function createProportionalFinancialEntries($entityParams, $lineIt
     }
   }
 
+  /**
+   * Function to replace contribution tokens.
+   *
+   * @param array $contributionIds
+   *
+   * @param string $subject
+   *
+   * @param array $subjectToken
+   *
+   * @param string $text
+   *
+   * @param string $html
+   *
+   * @param array $messageToken
+   *
+   * @param bool $escapeSmarty
+   *
+   * @return array
+   */
+  public static function replaceContributionTokens(
```

Comment:
@Edzelopez : here's what I think @eileenmcnaughton meant: elsewhere in code when rendering these contribution tokens (assuming that is supported somewhere), what function does the replacement of the tokens with their values.